src/main/scala/com/workday/warp/persistence/mysql/HasWarpSlickDsl.scala

```diff
@@ -52,6 +52,72 @@ trait HasWarpSlickDsl {
       expression.apply(timestamp)
     }
 
+    /**
+      *  Translates to YEAR() function
+      *  @return just the year in Int
+      */
+    def year(): Rep[Int] = {
+      val expression = SimpleExpression.unary[Timestamp, Int] { (timestamp, queryBuilder) =>
+        queryBuilder.sqlBuilder += " YEAR ("
+        queryBuilder.expr(timestamp)
+        queryBuilder.sqlBuilder += ")"
+      }
+      expression.apply(timestamp)
+    }
+
+    /**
+      * Translates to DATE() function
+      * @return just date in "yyyy-MM-dd"
+      */
+    def date(): Rep[String] = {
+      val expression = SimpleExpression.unary[Timestamp, String] { (timestamp, queryBuilder) =>
+        queryBuilder.sqlBuilder += " DATE ("
+        queryBuilder.expr(timestamp)
+        queryBuilder.sqlBuilder += ")"
+      }
+      expression.apply(timestamp)
+    }
+
+    /**
+      * UNIX_TIMESTAMP operation - seconds elapsed since 1970-01-01 00:00:00 UTC
+      * @return seconds in int
+      */
+    def UNIX_TIMESTAMP(): Rep[Int] = {
+      val expression = SimpleExpression.unary[Timestamp, Int] { (timestamp, queryBuilder) =>
+        queryBuilder.sqlBuilder += " UNIX_TIMESTAMP ("
+        queryBuilder.expr(timestamp)
+        queryBuilder.sqlBuilder += ")"
+      }
+      expression.apply(timestamp)
+    }
+
+    /**
+      * subdate operation: date - interval
+      * @return timestamp date
+      */
+    def subdate(date: String, interval: String): Rep[String] = {
```

Comment:
Same thing here. Note that subdate has 3 different return types, depending on the `Date` type that's passed to it: https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_date-add 

If we want to cover all of them, we would have to either overloading these functions or see what slick's Type Matcher does out of the box. Please document the specific type of operation that is performed here and add TODOs for the rest.